peer/brontide.go

```diff
@@ -3895,3 +3914,101 @@ func (p *Brontide) sendLinkUpdateMsg(msg lnwire.Message) {
 	// continue processing message.
 	chanStream.AddMsg(msg)
 }
+
+// handleLinkUpdateMsg returns a boolean to indicate whether the given message
+// can be sent to the channel's message stream. If this message's channel is
+// still pending, it will be cached.
+func (p *Brontide) handleLinkUpdateMsg(msg lnwire.Message,
+	chanID lnwire.ChannelID) bool {
+
+	p.activeChanMtx.RLock()
+	defer p.activeChanMtx.RUnlock()
+
+	channel, ok := p.activeChannels[chanID]
+
+	// If the channel doesn't exist, this indicates an error.
+	if !ok {
+		p.log.Errorf("Received msg %s for non-existing channel %v",
+			msg.MsgType(), chanID)
+		return false
+	}
+
+	// Return true if found the active channel.
+	if channel != nil {
+		return true
+	}
+
+	// Otherwise this is a pending channel, we will save the message to
+	// cache and reprocess it once the channel becomes active.
+	//
+	// Load the cache.
+	cache, ok := p.earlyMsgCaches[chanID]
+	if !ok {
+		// Create the cache for this specific channel.
+		cache = newEarlyMsgCache(maxEarlyMsgs)
+		p.earlyMsgCaches[chanID] = cache
+	}
+
+	// Create a cached message and save it.
+	cachedMsg := &cachedEarlyMsg{
+		msg: msg,
+	}
+	err := cache.addMsg(cachedMsg)
+	if err != nil {
+		p.log.Errorf("Failed to cache msg %s for pending channel: %v, "+
+			"error: %v", msg.MsgType(), chanID, err)
+	}
+
+	return false
+}
+
+// processEarlyMsgs takes a block height, resends all the future messages
+// found below and equal to that height and deletes those messages found in the
+// gossiper's futureMsgs.
```

Comment:
This function comment seems incorrect.  I don't see any block heights used or deleting of messages.

Reply:
yikes, it was copied, now fixed.